sim/src/index/componentType.h

```diff
@@ -0,0 +1,62 @@
+#ifndef BENTOBOX_COMPONENTTYPEINDEX_H
+#define BENTOBOX_COMPONENTTYPEINDEX_H
+
+#include <unordered_map>
+#include <functional>
+#include <core/ics/component.h>
+#include <core/ics/componentSet.h>
+
+namespace ics::index {
+    class ComponentType {
+    private:
+        std::unordered_map<std::type_index, size_t> map;
```

Comment:
Nitpick: `map` is not a very descriptive name. Maybe `typeIndexMap`

Reply:
Resolved in 0dba137